ui/src/org/safe.ts

```diff
@@ -10,11 +10,32 @@ import SafeServiceClient, {
   SafeMultisigTransactionResponse,
 } from "@gnosis.pm/safe-service-client";
 
+import { sleep } from "ui/src/sleep";
 import * as Ethereum from "ui/src/ethereum";
 import * as error from "ui/src/error";
 import type { Wallet } from "ui/src/wallet";
 import type { OperationType } from "@gnosis.pm/safe-core-sdk-types";
 
+export async function waitUntilSafeIsReady(
+  safeAddress: string,
+  ethEnv: Ethereum.Environment
+): Promise<void> {
+  safeAddress = ethers.utils.getAddress(safeAddress);
+  const safeServiceClient = createSafeServiceClient(ethEnv);
+  let tries = 360; // 30 min
+
+  while (tries > 0) {
```

Comment:
We can use `retryOnError` instead of rolling our own custom logic. I’ll fix this in a PR